apps/console/app/routes/_shell.agent.$agentSlug.branch.$branchSlug.config/form.tsx

```diff
@@ -0,0 +1,270 @@
+"use client";
+
+import { useActionData, useNavigation, useParams, useRouteLoaderData } from "react-router";
+import { useForm, getFormProps, getInputProps } from "@conform-to/react";
+import { parseWithValibot, getValibotConstraint } from "@conform-to/valibot";
+import { object, array, string, nonEmpty, pipe, trim, type InferOutput } from "valibot";
+
+import supportedModels from "@hebo/shared-data/json/supported-models";
+
+import { Button } from "@hebo/shared-ui/components/Button";
+import { Card, CardContent, CardFooter } from "@hebo/shared-ui/components/Card";
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@hebo/shared-ui/components/Collapsible";
+import { Input } from "@hebo/shared-ui/components/Input";
+import { Select } from "@hebo/shared-ui/components/Select";
+import { CopyToClipboardButton } from "@hebo/shared-ui/components/code/CopyToClipboardButton";
+import { Split } from "lucide-react";
+
+import { useActionDataErrorToast } from "~console/lib/errors";
+import { FormControl, FormField, FormLabel, FormMessage } from "@hebo/shared-ui/components/Form";
+import { Form } from "react-router";
+import { useState } from "react";
+
+export const ModelConfigSchema = object({
+  models: array(
+    object({
+      alias: pipe(string(), trim(), nonEmpty("Alias is required")),
+      type: pipe(string(), trim(), nonEmpty("Model type is required")),
+    })
+  ),
+});
+
+export type ModelConfigFormValues = InferOutput<typeof ModelConfigSchema>;
+
+const getModelDisplayName = (modelName: string): string => {
+  if (modelName === "custom") return "Custom Model";
+  const model = supportedModels.find((m) => m.name === modelName);
+  return model?.displayName || modelName;
+};
+
+export default function ModelConfigurationForm() {
+  const { agent } = useRouteLoaderData<{
+    agent: { branches: Array<{ models: Array<{ alias: string; type: string }> }> };
+  }>("routes/_shell.agent.$agentSlug")!;
+  
+  const { agentSlug, branchSlug } = useParams<{ agentSlug: string; branchSlug: string }>();
```

Comment:
Convention from other forms is to pass required data as props instead of using useRouteLoaderData.